### PR TITLE
Tweak REST-based client creation methods

### PR DESCRIPTION
**Pull request by jskeet**
The first commit is just for storage - once we're all happy with how that looks, I'll add further commits for BigQuery and Translate.

### CODE REVIEW

**Review by amanda-tarafa — APPROVED** [review]
Looks good, just a few comments.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs [hunk 1]
```diff
@@ -48,8 +45,6 @@ public abstract partial class StorageClient : IDisposable
     {
         private static readonly string[] s_scopes = new[] { StorageService.Scope.DevstorageFullControl };
 
-        private const string EndpointOverride = "https://storage.googleapis.com/storage/v1/";
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs:51]: Yes, it used to be https://www.googleapis.com/storage/v1/

This was one of the first APIs where we used a specific host, IIRC.

File: apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs [hunk 2]
```diff
@@ -71,40 +66,59 @@ public abstract partial class StorageClient : IDisposable
         /// </summary>
         public virtual EncryptionKey EncryptionKey { get { throw new NotImplementedException(); } }
 
+        /// <summary>
+        /// Asynchronously creates a <see cref="StorageClient"/> using application default credentials.
+        /// For any non-default values, please use <see cref="StorageClientBuilder"/>.
+        /// </summary>
+        /// <returns>The task representing the created <see cref="StorageClient"/>.</returns>
+        public static Task<StorageClient> CreateAsync() => new StorageClientBuilder().BuildAsync();
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClient.cs:74]: Nope, there's no ambiguity. When there are two overloads that are both applicable, if one is applicable without using any default argument values, that one is "better" than the one which is only applicable after applying default argument values.

**Comment by jskeet** [comment]
Changed "apply" to "use" in all four suggested locations.

**Comment by jskeet** [comment]
Having looked at BigQuery, it already has quite comprehensive comments on this, so I'll leave it alone. (We *could* add overloads that accept just a project ID, but it's probably simpler not to...)

Ready for final review.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Review by Rishabh-V — APPROVED** [review]
LGTM

**Pull request merged**